lib/Doctrine/DBAL/Driver/PDOSqlsrv/Driver.php

```diff
@@ -61,16 +66,56 @@ private function _constructPdoDsn(array $params)
         }
 
         if (isset($params['dbname'])) {
-            $dsn .= ';Database=' .  $params['dbname'];
+            $connectionOptions['Database'] = $params['dbname'];
         }
 
         if (isset($params['MultipleActiveResultSets'])) {
-            $dsn .= '; MultipleActiveResultSets=' . ($params['MultipleActiveResultSets'] ? 'true' : 'false');
+            $connectionOptions['MultipleActiveResultSets'] = $params['MultipleActiveResultSets'] ? 'true' : 'false';
         }
 
+        $dsn .= $this->getConnectionOptionsDsn($connectionOptions);
+
         return $dsn;
     }
 
+    /**
+     * Separates a connection options from a driver options
+     *
+     * @param mixed[] $options
+     * @return mixed[][]
```

Comment:
I know @Majkl578 won't like these `mixed` annotation. Taking into account that most existing tools will not recognize annotations like `(int|string)[]`, would it be fair to replace them with `int[]|string[]` and `int[][]|string[][]` respectively?

Reply:
Probably, better than mixed. :+1: